### PR TITLE
Add Retry to vswhere download

### PR DESCRIPTION
**Pull request by sunandabalu**
Fixes https://github.com/dotnet/arcade/issues/6657

### CODE REVIEW

File: eng/common/tools.ps1 [hunk 1]
```diff
@@ -439,11 +439,26 @@ function LocateVisualStudio([object]$vsRequirements = $null){
...
+        Write-PipelineTelemetryError -Category 'InitializeToolset' -Message $_
+      }
+
+      if (++$retries -le $maxRetries) {
+        $delayInSeconds = [math]::Pow(2, $retries) - 1 # Exponential backoff
+        Write-Host "Retrying. Waiting for $delayInSeconds seconds before next attempt ($retries of $maxRetries)."
+        Start-Sleep -Seconds $delayInSeconds
+      }
+      else {
+        throw "Unable to download file in $maxRetries attempts."
```
Comment (MattGal) [inline eng/common/tools.ps1:460]: This throw is new, we probably still want to Write-PipelineTelemetryError here.

Comment (sunandabalu) [reply eng/common/tools.ps1]: Changed.

**Review by missymessa — APPROVED** [review]
LGTM shipit

File: eng/common/tools.ps1 [hunk 2]
```diff
@@ -439,11 +439,26 @@ function LocateVisualStudio([object]$vsRequirements = $null){
...
+    while($true) {
+      try {
+        Invoke-WebRequest "https://netcorenativeassets.blob.core.windows.net/resource-packages/external/windows/vswhere/$vswhereVersion/vswhere.exe" -OutFile $vswhereExe
+        break
+      }
+      catch{
+        Write-PipelineTelemetryError -Category 'InitializeToolset' -Message $_
+      }
+
+      if (++$retries -le $maxRetries) {
```
Comment (riarenas) [inline eng/common/tools.ps1:454]: Doesn't this pre-increment mean we're only doing this 4 times in total?

Comment (riarenas) [reply eng/common/tools.ps1:454]: nevermind, I missed this was -le and not -lt.

**Pull request merged**